cms/djangoapps/export_course_metadata/signals.py

```diff
@@ -0,0 +1,40 @@
+"""
+This file exports metadata about the course.
+Initially the export goes to S3,
+but the storage class could be used for other backends in the future.
+
+This data is initially being used by Braze Connected Content to include
+section highlights in emails, but may be used for other things in the future.
```

Comment:
Similarly to what MT said before about talking about Braze in this code. Not sure it adds value or is relevant for the broader community. The idea behind this is just your first sentence about exporting course metadata. The way it's used is up to whoever.

Reply:
well doesn't our documentation have two consumers, the openedx community but also the edx.org employees?  
for the edx.org people wouldn't it be valuable to know what code braze is depending on?  
for instance, maybe i'm someone who needs to make a change to a braze integration and I want to be able to search braze in edx-platform and find the right spots?

Reply:
(but also if we care about openedx for this feature wouldn't we'd want some sort of release notes or something that are like, hey we're exporting highlights to an s3 bucket if you define the bucket, without those docs they would presumably not know anything about using this feature anyways?)

Reply:
I think it's a matter of where you document it and how you phrase it. 

If an open edx installation filed a PR to add a comment here that said "these highlights are also used for sailthru by XXX company" -- that would feel weird right?

Maybe an internal marketing wiki page to document how we use this export data would make more sense, if you are thinking we want to keep track? And/or just make your code comments vaguer, like "this could be used by an external service like Braze to send highlights" so it's a little more general but still clues in developers looking at this on how it's likely used?

Reply:
> (but also if we care about openedx for this feature wouldn't we'd want some sort of release notes or something that are like, hey we're exporting highlights to an s3 bucket if you define the bucket, without those docs they would presumably not know anything about using this feature anyways?)

I mean, yeah. This seems like something that would be suitable for mentioning in release notes. Do you think we shouldn't?

But even if we didn't mention it in the release notes, I don't see why making a feature hard-to-discover means we should put edx.org internal process documentation next to the feature.

Reply:
i was thinking more of a use case like we decide in a year that we want to go shut down the braze integration because the extra expense of braze isn't worth it, so then some employee searches braze in the code to find the things braze depends on to delete. 

(or we want to find the right spots in the braze integration to deprecate or something)
i tend to prefer colocating documentation with the code where possible since almost all of the multi-year-old wiki pages i've read have been super outdated
i do like your suggestion though of making the comments slightly vaguer by using the word could and changing the tense so i'll do that

Reply:
I think I'm just of the opinion that that is independent of the feature. The feature is uploading to S3. Nobody but edX needs to care about Braze. Even the vague language is odd because again, any consumer could use it anyway they want. It's just JSON in an S3 bucket. We don't inform people sample use cases for hitting our API.

And I get what you're saying about wanting to be able to search Braze and know what's connected to it and using it, but that's on us as a company. Not something we should also impose on our consumers. If we need a private doc for our team to keep up with that, then sure. But it feels wrong to me to force our documentation onto the community. If this resource could _only_ be consumed by Braze, then sure. But it can be consumed in numerous ways so I disagree with putting in our own logic.

Reply:
well the feature isn't uploading to s3, it is saving highlights to filestorage or s3 or a different backend if they configure a different backend in the settings.  
but also, even though we have plenty of edx/sailthru specific comments in edx-platform, I don't see why that's necessarily a bad thing.  
if any openedx customers were hypothetically using our sailthru integration, they could also hypothetically use our braze integration if they decide to pay for braze in which case this could be helpful?  
though i'm a bit confused, if we want to support this feature for openedx, per my above comment wouldn't we want a doc describing this feature and include this feature in the release notes and so on - that seems to me like it would be the main thing if we want to make sure openedx knows how to use this

Reply:
Sorry, was specifically referring to the S3Boto3 storage backend where it is S3 specific. Totally agree it works for all sorts of backends.

I totally agree with you about adding this into release notes for the general community. Maybe something like 'Added a new export-course-metadata-to-storage feature. Toggle it with _. Related settings are _, _, and _. Useful for external services you might have that want to scrape course data.'

Reply:
where do the release notes live?  

well, it is just highlights, not course data?  
but if we thought we want openedx scrapers to run against this for retrieving course data this seems like it would be very janky and things should be piped through discovery instead?

Reply:
This is the running tally of "stuff we should say when releasing lilac": https://openedx.atlassian.net/wiki/spaces/COMM/pages/2023915819/Lilac - anyone can add to it

Reply:
Well it is highlights right now, but could be amended. Seems nice to just leave that part abstract since it technically just supports any amount of course data pushed to a storage backend. 

And we aren't going to be advertising our private buckets for use by openedx scrapers. When I said external services could scrape course data, I meant that more in the sense of third party services you allow. So like in our case, it would be Braze, but generically could be hit by anyone with access to your storage

Reply:
added a change to the release notes, will also just remove the braze comment here
I don't really like this solution but I'll just bring up some of my concerns on Jason's doc